src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -144,358 +58,296 @@ export class ZHANetworkVisualizationPage extends LitElement {
     }
-    if (device.device_type === "Coordinator") {
-      return 2;
-    }
-    if (device.device_type === "Router") {
-      return 4;
-    }
-    return 5;
-  }
-
-  private _getShape(device: ZHADevice): string {
-    if (device.device_type === "Coordinator") {
-      return "box";
+    const device = this._devices.find((d) => d.ieee === (data as any).id);
+    if (!device) {
+      return name;
     }
-    if (device.device_type === "Router") {
-      return "ellipse";
-    }
-    return "circle";
-  }
-
-  private _buildLabel(device: ZHADevice): string {
-    let label =
-      device.user_given_name !== null
-        ? `<b>${device.user_given_name}</b>\n`
-        : "";
-    label += `<b>IEEE: </b>${device.ieee}`;
-    label += `\n<b>Device Type: </b>${device.device_type.replace("_", " ")}`;
+    let label = `<b>IEEE: </b>${device.ieee}`;
+    label += `<br><b>Device Type: </b>${device.device_type.replace("_", " ")}`;
     if (device.nwk != null) {
-      label += `\n<b>NWK: </b>${formatAsPaddedHex(device.nwk)}`;
+      label += `<br><b>NWK: </b>${formatAsPaddedHex(device.nwk)}`;
     }
     if (device.manufacturer != null && device.model != null) {
-      label += `\n<b>Device: </b>${device.manufacturer} ${device.model}`;
+      label += `<br><b>Device: </b>${device.manufacturer} ${device.model}`;
     } else {
-      label += "\n<b>Device is not in <i>'zigbee.db'</i></b>";
+      label += "<br><b>Device is not in <i>'zigbee.db'</i></b>";
     }
     if (device.area_id) {
-      label += `\n<b>Area ID: </b>${device.area_id}`;
-    }
-    return label;
-  }
-
-  private _handleSearchChange(ev: CustomEvent) {
-    this._filter = ev.detail.value;
-    const filterText = this._filter!.toLowerCase();
-    if (!this._network) {
-      return;
-    }
-    if (this._filter) {
-      const filteredNodeIds: (string | number)[] = [];
-      this._nodes.forEach((node) => {
-        if (node.label && node.label.toLowerCase().includes(filterText)) {
-          filteredNodeIds.push(node.id!);
-        }
-      });
-      this.zoomedDeviceId = "";
-      this._zoomOut();
-      this._network.selectNodes(filteredNodeIds, true);
-    } else {
-      this._network.unselectAll();
-    }
-  }
-
-  private _onZoomToDevice(event: ValueChangedEvent<string>) {
-    event.stopPropagation();
-    this.zoomedDeviceId = event.detail.value;
-    if (!this._network) {
-      return;
-    }
-    this._zoomToDevice();
-  }
-
-  private _zoomToDevice() {
-    this._filter = "";
-    if (!this.zoomedDeviceId) {
-      this._zoomOut();
-    } else {
-      const device: ZHADevice | undefined = this._devicesByDeviceId.get(
-        this.zoomedDeviceId
-      );
-      if (device) {
-        this._network!.fit({
-          nodes: [device.ieee],
-          animation: { duration: 500, easingFunction: "easeInQuad" },
-        });
+      const area = this.hass.areas[device.area_id];
+      if (area) {
+        label += `<br><b>Area: </b>${area.name}`;
```

Comment:
@MindFreeze Should we add translations here?

Reply:
https://github.com/home-assistant/frontend/pull/25457#issuecomment-2889729140